### PR TITLE
✨ Simple reactor example

### LINKED ISSUE #760: Write documentation for builders and designs
## Description of issue / requirement to address

We currently have examples and API documentation but no user guides around designs and builders. We need to add something in describing the intent of those methods, how to use them, and how to extend them.

### LINKED ISSUE #1537: Recreate design shape and plasma simple examples
## Description of issue / requirement to address

The design shape and design plasma examples will be removed in #1536 (see #1523) 

## Proposed solution

Create some simple design examples

## Alternative solutions

None

## Additional context

Removed as no longer works in current form

### PR DESCRIPTION
**Pull request by je-cook**
## Linked Issues

Expanded from #1229 

Closes #1537 
Closes #760 when combined with #1567 
Part of #1073

## Description

A simple reactor

## Interface Changes

None

## Checklist

I confirm that I have completed the following checks:

- [ ] Tests run locally and pass `pytest tests --reactor`
- [ ] Code quality checks run locally and pass `flake8` and `black .`
- [ ] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

**Comment by je-cook** [comment]
@CoronelBuendia and @ivanmaione this is the simple reactor example working as we started in #1229. Let 'e know if you want any changes

File: examples/design/simple_reactor.py [hunk 1]
```diff
@@ -0,0 +1,547 @@
...
+            self.build_config["param_class"],
+            default_module="bluemira.geometry.parameterisations",
+        )
+
+    def run(self) -> GeometryParameterisation:
+        """TF coil run method"""
+        parameterisation = self.parameterisation_cls(
+            var_dict={
+                "x1": {"value": 3.0, "fixed": True},
+                "x2": {"value": 15, "lower_bound": 12},
```
Comment (hsaunders1904) [inline examples/design/simple_reactor.py:337]: Won't this only work for a `PrincetonD`? What happens if we pass in another geometry parameterisation? Isn't this `var_dict` usually defined in the `build_config`?

File: examples/design/simple_reactor.py [hunk 2]
```diff
@@ -0,0 +1,547 @@
...
+}
+
+
+# %%[markdown]
+
+# Now we set up our ParamterFrames
+
+# %%
+
+# TODO improve build config manipulation
```
Comment (hsaunders1904) [inline examples/design/simple_reactor.py:502]: Big +1 on this. If there isn't one, we should get an issue opened for this for the v1 milestone. There's probably a fair bit of discussion to have on how we do this manipulation in the most sensible way

File: examples/design/simple_reactor.py [hunk 3]
```diff
@@ -0,0 +1,547 @@
...
+        component.display_cad_options.color = BLUE_PALETTE["PL"]
+        component.display_cad_options.transparency = 0.5
+        return component
+
+
+# %%[markdown]
+
+# And now the TF Coil, in this instance for simplicity we are only making one TF coil.
+# If more TF coils were to be required the build_xyz of `TFCoilBuilder` would need to
+# be modified.
```
Comment (hsaunders1904) [inline examples/design/simple_reactor.py:313]: We should give a quick run-through of exactly what this designer is doing. It's finding the given geometry parameterisation, then feeding that into the optimisation problem we made earlier, then running the optimisation problem to generate the centreline wire

File: examples/design/simple_reactor.py [hunk 4]
```diff
@@ -0,0 +1,547 @@
...
+class MyReactor(Reactor):
+    """Reactor container"""
+
+    plasma: Plasma
+    tf_coil: TFCoil
+
+
+# %%[markdown]
+
+# Now we want to define a way to optimise the TF coil shape.
```
[the remark this answers is not in the document]
Comment (je-cook) [reply examples/design/simple_reactor.py:138]: Done

File: examples/design/simple_reactor.py [hunk 5]
```diff
@@ -0,0 +1,547 @@
...
+
+    def lcfs(self):
+        """Get separatrix"""
+        return (
+            self.component().get_component("xz").get_component("LCFS").shape.boundary[0]
+        )
+
+
+class TFCoil(ComponentManager):
+    """TF Coil manager"""
```
[the remark this answers is not in the document]
Comment (CoronelBuendia) [reply examples/design/simple_reactor.py:118]: Agreed, I threw something in for this

**Review by CoronelBuendia — APPROVED** [review]
I think this is a lot cleaner than what we had, so good job getting this all to work guys! It's also a good simple example for somebody wanting to start something from scratch. I'm happy with it as is, but I'm keen to get the EU-DEMO thing back up and running to see if we've missed anything in terms of connections between the different abstractions.

**Review by hsaunders1904 — APPROVED** [review]

**Pull request merged**